src/components/ConfirmedRoute.js

```diff
@@ -0,0 +1,119 @@
+    transaction: {},
+    mapboxToken: '',
+};
+
+const getWaypointMarkers = (waypoints) => {
+    const numberOfWaypoints = _.size(waypoints);
+    const lastWaypointIndex = numberOfWaypoints - 1;
+    return _.filter(
+        _.map(waypoints, (waypoint, key) => {
+            if (!waypoint || waypoint.lng === undefined || waypoint.lat === undefined) {
+                return;
+            }
+
+            const index = Number(key.replace('waypoint', ''));
+            let MarkerComponent;
+            if (index === 0) {
+                MarkerComponent = Expensicons.DotIndicatorUnfilled;
+            } else if (index === lastWaypointIndex) {
+                MarkerComponent = Expensicons.Location;
+            } else {
+                MarkerComponent = Expensicons.DotIndicator;
+            }
+
+            return {
+                coordinate: [waypoint.lng, waypoint.lat],
+                markerComponent: () => (
+                    <MarkerComponent
+                        width={20}
+                        height={20}
+                        fill={theme.icon}
+                    />
+                ),
+            };
+        }),
+        (waypoint) => waypoint,
+    );
+};
+
+function ConfirmedRoute({mapboxToken, transaction}) {
+    const {isOffline} = useNetwork();
+    const {translate} = useLocalize();
+    const {route0: route} = transaction.routes;
+    const waypoints = lodashGet(transaction, 'comment.waypoints', {});
+    const waypointMarkers = getWaypointMarkers(waypoints);
+
+    useEffect(() => {
+        MapboxToken.init();
+        return MapboxToken.stop;
+    }, []);
+
+    return (
+        <>
+            {!isOffline && mapboxToken ? (
+                <MapView
+                    accessToken={mapboxToken}
+                    mapPadding={MAP_PADDING}
+                    pitchEnabled={false}
+                    directionCoordinates={route.geometry.coordinates}
+                    directionStyle={styles.mapDirection}
+                    style={styles.mapView}
+                    waypoints={waypointMarkers}
+                    styleURL={CONST.MAPBOX_STYLE_URL}
+                />
+            ) : (
+                <View style={[styles.mapPendingView]}>
+                    <BlockingView
+                        icon={Expensicons.EmptyStateRoutePending}
+                        title={translate('distance.mapPending.title')}
+                        subtitle={isOffline ? translate('distance.mapPending.subtitle') : translate('distance.mapPending.onlineSubtitle')}
+                        shouldShowLink={false}
+                    />
+                </View>
+            )}
+        </>
+    );
+}
+
+export default withOnyx({
+    transaction: {
+        key: ({transactionID}) => `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`,
+    },
+    mapboxToken: {
+        key: ONYXKEYS.MAPBOX_ACCESS_TOKEN,
+        selector: (mapboxAccessToken) => (mapboxAccessToken ? mapboxAccessToken.token : ''),
```

Comment:
```suggestion
        selector: (mapboxAccessToken) => mapboxAccessToken.token,
```
The ternary isn't necessary. If the `selector` returns `undefined` then the `defaultProps` will work. With this ternary, it is making the `defaultProps` obsolete.